Libplanet/Store/DefaultStore.cs

```diff
@@ -246,174 +271,237 @@ public override IEnumerable<TxId> IterateStagedTransactionIds()
+                    _txs.WriteAllBytes(tmpPath, txBytes);
+                    try
+                    {
+                        _txs.MoveFile(tmpPath, path);
+                    }
+                    catch (IOException)
+                    {
+                        // For the same txid, the content is the same as well.  Don't have
+                        // to be written more than once.
+                        if (!_txs.FileExists(path) ||
+                            _txs.GetFileLength(path) != txBytes.LongLength)
+                        {
+                            throw;
+                        }
+                    }
                 }
                 finally
                 {
-                    _txLock.ExitWriteLock();
+                    if (_txs.FileExists(tmpPath))
+                    {
+                        _txs.DeleteFile(tmpPath);
+                    }
                 }
             }
-            finally
-            {
-                _txLock.ExitUpgradeableReadLock();
-            }
+
+            _txCache.AddOrUpdate(tx.Id, tx);
         }
 
         /// <inheritdoc/>
         public override bool DeleteTransaction(TxId txid)
         {
-            _txLock.EnterWriteLock();
-            try
+            var path = TxPath(txid);
+            if (_txs.FileExists(path))
             {
-                return _db.FileStorage.Delete(TxFileId(txid));
-            }
-            finally
-            {
-                _txLock.ExitWriteLock();
+                _txs.DeleteFile(path);
+                _txCache.Remove(txid);
+                return true;
             }
+
+            return false;
         }
 
         /// <inheritdoc/>
         public override IEnumerable<HashDigest<SHA256>> IterateBlockHashes()
         {
-            _blockLock.EnterReadLock();
-            try
+            foreach (UPath path in _blocks.EnumerateDirectories(UPath.Root))
             {
-                return _db.FileStorage
-                    .Find("block/")
-                    .Select(file => new HashDigest<SHA256>(ByteUtil.ParseHex(file.Filename)));
-            }
-            finally
-            {
-                _blockLock.ExitReadLock();
+                string upper = path.GetName();
+                if (upper.Length != 2)
+                {
+                    continue;
+                }
+
+                foreach (UPath subPath in _blocks.EnumerateFiles(path))
+                {
+                    string lower = subPath.GetName();
+                    if (lower.Length != 62)
+                    {
+                        continue;
+                    }
+
+                    string name = upper + lower;
+                    HashDigest<SHA256> blockHash;
+                    try
+                    {
+                        blockHash = new HashDigest<SHA256>(ByteUtil.ParseHex(name));
+                    }
+                    catch (Exception)
+                    {
+                        continue;
```

Comment:
Is this `catch` intended? if so, it would be better that we describe comments about why we should catch them.

Reply:
Fixed: d5dadb3e4532dbbe132f9e85832461ad34cbef05.